File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/map/BaseSymbolController.java
package com.valentingrigorean.arcgis_maps_flutter.map;

public abstract class BaseSymbolController extends BaseSymbolWorkerController  implements SymbolsController{

    private SymbolVisibilityFilterController symbolVisibilityFilterController;
    private SelectionPropertiesHandler selectionPropertiesHandler;

    public SymbolVisibilityFilterController getSymbolVisibilityFilterController() {
        return symbolVisibilityFilterController;
    }

    public void setSymbolVisibilityFilterController(SymbolVisibilityFilterController symbolVisibilityFilterController) {
        this.symbolVisibilityFilterController = symbolVisibilityFilterController;
    }

    public SelectionPropertiesHandler getSelectionPropertiesHandler() {
        return selectionPropertiesHandler;
    }

    public void setSelectionPropertiesHandler(SelectionPropertiesHandler selectionPropertiesHandler) {
        this.selectionPropertiesHandler = selectionPropertiesHandler;
    }

    protected void onSymbolRemoval(GraphicControllerSink controller) {
        final SymbolVisibilityFilterController visibilityFilterController = getSymbolVisibilityFilterController();
        if (visibilityFilterController != null) {
            visibilityFilterController.removeGraphicsController(controller);
        }
    }

    protected void invalidateVisibilityFilterController(GraphicControllerSink controller) {
        final SymbolVisibilityFilterController filterController = getSymbolVisibilityFilterController();
        if (filterController != null) {
            filterController.invalidate(controller);
        }
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/map/BaseSymbolController.kt
package com.valentingrigorean.arcgis_maps_flutter.map

abstract class BaseSymbolController : SymbolsController {
    override var symbolVisibilityFilterController: SymbolVisibilityFilterController? = null
    override var selectionPropertiesHandler: SelectionPropertiesHandler? = null
    protected fun onSymbolRemoval(controller: GraphicControllerSink) {
        val visibilityFilterController = symbolVisibilityFilterController
        visibilityFilterController?.removeGraphicsController(controller)
    }

    protected fun invalidateVisibilityFilterController(controller: GraphicControllerSink) {
        val filterController = symbolVisibilityFilterController
        filterController?.invalidate(controller)
    }
}